src/rgw/rgw_pubsub.cc

```diff
@@ -582,9 +582,8 @@ int RGWPubSub::get_topics(const DoutPrefixProvider* dpp,
   }
 
   if (!use_notification_v2 || driver->stat_topics_v1(tenant, y, dpp) != -ENOENT) {
-    // in case of v1 or during migration we use v1 topics
-    // v1 returns all topics, ignoring marker/max_items
-    return read_topics_v1(dpp, result, nullptr, y);
+    // you must check if all zones are v2 ready and migration is done
```

Comment:
i think we can remove this validation. if `get_topics_v2()` we should not expect valid values if we call on v1 or during migration